src/data_vis.ts

```diff
@@ -326,155 +280,62 @@ class DataVisModel {
             selectedVaccines: this.vaccineFilter().selectedOptions(), // which vaccines do we care about
             selectedTouchstones: this.touchstoneFilter().selectedOptions(), // which touchstones do we care about
             cumulative: (this.compare() == "year" && this.cumulativePlot()), // are we creating a cumulative plot
-            timeSeries: false
-        };
+            timeSeries: this.currentPlot() == "Time series",
+            yAxisTitle: this.yAxisTitle(),
+            plotTitle: this.plotTitle(),
+            hideLabels: this.hideLabels()
+        }
+
+    }, this).extend({rateLimit: 250});
```

Comment:
`this.chartOptions` is an object that contains all the info needed for the charts in one place. Because it's computed, whenever a sub-property changes it will automatically update, but I've also added a rate limiter which means it will only fire it's own change event every 250ms at most. This means that if we listen to changes on this object, changes to the sub-properties will be batched together.